Added Liquidity Ratio Support

ashi20192020ayush opened the pull request:
resolves #471

drbh commented on pallets/amm/src/lib.rs:
```diff
@@ -156,6 +159,38 @@ pub mod pallet {
+                    if optimal_quote_amount <= quote_amount {
+                        (base_amount, optimal_quote_amount)
+                    } else {
+                        let optimal_base_amount = Self::quote(
+                            quote_amount,
+                            pool_liquidity_amount.base_amount,
+                            pool_liquidity_amount.quote_amount,
+                        );
```
I think the base and quote amounts need to be inverse from the `optimal_quote_amount` in order to calculate `optimal_base_amount`


```rust
let optimal_base_amount = Self::quote(
    quote_amount,
    pool_liquidity_amount.quote_amount,
    pool_liquidity_amount.base_amount,
);
```

[similar to uniswap](https://github.com/Uniswap/uniswap-v2-periphery/blob/dda62473e2da448bc9cb8f4514dadda4aeede5f4/contracts/UniswapV2Router02.sol#L54)

ETeissonniere replied:
I second that, this seems correct

ETeissonniere commented on pallets/amm/src/lib.rs:
```diff
@@ -156,6 +159,38 @@ pub mod pallet {
+                        (optimal_base_amount, quote_amount)
+                    };
+
+                let (minimum_base_amount, minimum_quote_amount) = match is_inverted {
+                    true => (minimum_amounts.1, minimum_amounts.0),
+                    false => (minimum_amounts.0, minimum_amounts.1),
+                };
```
I think this can actually be rewritten to this
```suggestion
                let (minimum_base_amount, minimum_quote_amount) = if is_inverted {
                    (minimum_amounts.1, minimum_amounts.0)
                } else {
                    (minimum_amounts.0, minimum_amounts.1)
                };
```

drbh commented:
this looks good! However, I think @ETeissonniere mentioned adding a unit test "to ensure / demo that we do select the correct ratio". Should we add that before merging?

ashi20192020ayush commented:
> this looks good! However, I think @ETeissonniere mentioned adding a unit test "to ensure / demo that we do select the correct ratio". Should we add that before merging?

@ETeissonniere @drbh I think this is already covered in existing test-cases because I modified existing test-cases to execute successfully. I also added one negative scenario https://github.com/parallel-finance/parallel/blob/258716f7b6a4e515d60187899673c9b409335ec4/pallets/amm/src/tests.rs#L56

Please provide some details about new test-case scenario.
